web/packages/teleterm/src/ui/ConnectMyComputer/DocumentConnectMyComputer/Setup.tsx

```diff
@@ -411,14 +399,7 @@ function AgentSetup() {
   };
 
   useEffect(() => {
-    if (
-      [
-        createRoleAttempt,
-        downloadAgentAttempt,
-        generateConfigFileAttempt,
-        joinClusterAttempt,
-      ].every(attempt => attempt.status === '')
-    ) {
+    if (steps.every(step => step.attempt.status === '')) {
```

Comment:
I just encountered a bug where `Setup` page stuck in an invalid state.
If the setup fails, for example on joining the cluster, then you close the tab and open it again, you will see the stuck setup:
<img width="829" alt="image" src="https://github.com/gravitational/teleport/assets/20583051/bdb36f88-55c3-4e4a-be06-cba3bffe4a21">

It happens because `downloadAgentAttempt` is kept in the context, so it doesn't reset when the tab is closed. 
I think we can get rid of the condition, `useEffect` doesn't have any deps and we only want to run it when the component mounts.
Another way to fix this would be to reorganize the `Setup` component, so clicking on the `Connect` button would trigger the actual setup process, but this is too much work.

Reply:
Right, I've run into this as well. How would we address this if strict mode was on?

Reply:
It seems to me like the correct solution would be to abort a controller on unmount + reset the download attempt to an empty attempt.

Or even, maybe Connect shouldn't re-use the attempt from the context. It should have it's own attempt which merely runs `runDownloadAgentAttempt`. We'd still need to abort any operation on abort though. I'm not sure if we ever done that over IPC, but it shouldn't be much different than over gRPC.

Reply:
>How would we address this if strict mode was on?

Good point 🤔

>Or even, maybe Connect shouldn't re-use the attempt from the context. It should have its own attempt which merely runs runDownloadAgentAttempt. 

That sounds like a better option to me, I wouldn't reset the context's attempt.

>We'd still need to abort any operation on abort though. I'm not sure if we ever done that over IPC, but it shouldn't be much different than over gRPC.

I'm afraid it's not possible to abort an async IPC operation (at least quick googling didn't return anything) :/

Reply:
> I'm afraid it's not possible to abort an async IPC operation (at least quick googling didn't return anything) :/

There's no generic way to abort an IPC call because aborting an IPC call is not quite the correct way to think about it. IIRC the way to do it is to pass an "objectified" signal over the context bridge. Then the underlying operation in the other process needs to support cancellation in some way.

Reply:
In any case, I think for now I'm going to remove the deps and leave a TODO comment about making this code more strict-mode-friendly.